gcapi/client.py

```diff
@@ -845,25 +847,61 @@ def update_archive_item(
             )
         )
 
-    def create_display_sets_from_images(
-        self, *, reader_study: str, display_sets: List[Dict[str, list]]
+    def _validate_display_set_values(self, values):
```

Comment:
If you create a set of the slugs, you only need to check once for every interface. If you keep a dict of retrieved interfaces, you can reduce the nr of calls even further, as now the interface is retrieved in this function, and again in create_display_sets_from_values.

Reply:
I think this covers what you mean: https://github.com/DIAGNijmegen/rse-gcapi/pull/136/commits/eb539f3eed7be66ec194a64183514f9f51ed586c :slightly_smiling_face: